src/OpenLoco/src/Vehicles/VehicleManager.cpp

```diff
@@ -119,21 +123,146 @@ namespace OpenLoco::VehicleManager
+            const auto newPos = World::Pos3{ base.tileX, base.tileY, base.tileBaseZ * World::kSmallZStep } + moveInfo.loc;
+            base.spriteYaw = moveInfo.yaw;
+            base.spritePitch = moveInfo.pitch;
+            base.moveTo(newPos);
+        }
+        base.invalidateSprite();
+    }
+
     // 0x004B05E4
     PlaceDownResult placeDownVehicle(Vehicles::VehicleHead* const head, const coord_t x, const coord_t y, const uint8_t baseZ, const Vehicles::TrackAndDirection& unk1, const uint16_t unk2)
     {
-        registers regs{};
-        regs.esi = X86Pointer(head);
-        regs.ax = x;
-        regs.cx = y;
-        regs.bx = unk2;
-        regs.dl = baseZ;
-        regs.ebp = unk1.track._data;
-        bool hasFailed = call(0x004B05E4, regs) & X86_FLAG_CARRY;
-        if (hasFailed)
+        const auto pos = World::Pos3{ x, y, baseZ * World::kSmallZStep };
+        if (head->tileX != -1)
+        {
+            return PlaceDownResult::Okay;
+        }
+
+        auto subPosition = 0;
+        World::Pos3 reversePos = pos;
+        Vehicles::TrackAndDirection reverseTad = unk1;
+        if (head->mode != TransportMode::road)
         {
-            return regs.al == 0 ? PlaceDownResult::Unk0 : PlaceDownResult::Unk1;
+            if (Vehicles::sub_4A2A58(pos, unk1.track, head->owner, head->trackType) & (1U << 0))
+            {
+                return PlaceDownResult::Unk1;
+            }
+
+            if (Vehicles::isBlockOccupied(pos, unk1.track, head->owner, head->trackType))
+            {
+                return PlaceDownResult::Unk1;
+            }
+
+            const auto subPositionLength = World::TrackData::getTrackSubPositon(unk1.track._data).size();
+            subPosition = subPositionLength - 1 - unk2;
+
+            const auto& trackSize = World::TrackData::getUnkTrack(unk1.track._data);
+            reversePos += trackSize.pos;
+            if (trackSize.rotationEnd < 12)
+            {
+                reversePos -= World::Pos3{ World::kRotationOffset[trackSize.rotationEnd], 0 };
+            }
+            reverseTad.track.setReversed(!unk1.track.isReversed());
         }
+        else
+        {
+
+            const auto subPositionLength = World::TrackData::getRoadSubPositon(unk1.road._data).size();
+            subPosition = subPositionLength - 1 - unk2;
+
+            const auto& roadSize = World::TrackData::getUnkRoad(unk1.road._data & 0x7F);
+            reversePos += roadSize.pos;
+            if (roadSize.rotationEnd < 12)
+            {
+                reversePos -= World::Pos3{ World::kRotationOffset[roadSize.rotationEnd], 0 };
+            }
+            reverseTad.road.setReversed(!unk1.road.isReversed());
+            reverseTad.road._data ^= (1U << 7);
+            if (reverseTad.road.isUnk8())
+            {
+                reverseTad.road._data ^= (1U << 7);
+                if (!reverseTad.road.isBackToFront())
+                {
+                    reverseTad.road._data ^= (1U << 8);
```

Comment:
Any idea what these flags represent?

Reply:
See the other pr i know what they are but going to clean it up after.